dali/ft/filecopy.cpp

```diff
@@ -830,7 +840,12 @@ void FileSprayer::beforeTransfer()
                     splitUNCFilename(remoteFilename.str(), &remoteDirectory, &remoteDirectory, NULL, NULL);
 
                     Owned<IFile> dir = createIFile(remoteDirectory.str());
+                    bool modPerms = false;
+                    if (!dir->exists() && (umask != -1))
+                        modPerms = true;
                     dir->createDirectory();
+                    if (modPerms)
+                        dir->setFilePermissions(~umask&0777);
```

Comment:
It would be cleaner if IFile::createDirectory() were extended to support mode, since mkdir that it ultimately calls does, thereby leaving no window where it has the wrong perms.